File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/repository/jpa/UserJpaRepository.java
package com.cakk.domain.mysql.repository.jpa;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.cakk.domain.mysql.entity.user.User;

public interface UserJpaRepository extends JpaRepository<User, Long> {

	Optional<User> findByProviderId(String providerId);
}

File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/repository/jpa/UserJpaRepository.kt
package com.cakk.domain.mysql.repository.jpa

import com.cakk.domain.mysql.entity.user.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserJpaRepository : JpaRepository<User, Long> {
    fun findByProviderId(providerId: String): User?
}
